torchvision/transforms/functional_tensor.py

```diff
@@ -44,22 +44,19 @@ def get_image_num_channels(img: Tensor) -> int:
     raise TypeError(f"Input ndim should be 2 or more. Got {img.ndim}")
 
 
-def _max_value(dtype: torch.dtype) -> float:
-    # TODO: replace this method with torch.iinfo when it gets torchscript support.
-    # https://github.com/pytorch/pytorch/issues/41492
-
-    a = torch.tensor(2, dtype=dtype)
-    signed = 1 if torch.tensor(0, dtype=dtype).is_signed() else 0
-    bits = 1
-    max_value = torch.tensor(-signed, dtype=torch.long)
-    while True:
-        next_value = a.pow(bits - signed).sub(1)
-        if next_value > max_value:
-            max_value = next_value
-            bits *= 2
-        else:
-            break
-    return max_value.item()
+def _max_value(dtype: torch.dtype) -> int:
+    if dtype == torch.uint8:
+        return int(2 ** 8) - 1
+    elif dtype == torch.int8:
+        return int(2 ** 7) - 1
+    elif dtype == torch.int16:
+        return int(2 ** 15) - 1
+    elif dtype == torch.int32:
+        return int(2 ** 31) - 1
+    elif dtype == torch.int64:
+        return int(2 ** 63) - 1
+    else:
+        return 1
```

Comment:
JIT on internal tests REALLY doesn't like this and fails with:
```
runtime error: 9.22337e+18 is outside the range of representable values of type 'long'
```

Replacing this with the following works:
```python
def _max_value(dtype: torch.dtype) -> int:
    if dtype == torch.uint8:
        return 255
    elif dtype == torch.int8:
        return 127
    elif dtype == torch.int16:
        return 32767
    elif dtype == torch.int32:
        return 2147483647
    elif dtype == torch.int64:
        return 9223372036854775807
    else:
        return 1
```

I'll cherrypick it after it's merged internally.